### ISSUE TITLE
Create TestPlan class

### ISSUE THREAD

**Issue by jamesrichford** [alsatian-test/alsatian#75]
To handle running the test set a test plan should be built from a given test plan e.g.

```

export class TestPlan {

   private _testFunctions: Array<(...args: Array<any>) => any>;

   public constructor(testSet: TestSet) {
       //
   }

   public run(): Promise<TestResults> { }
}

```

Given this change the responsibilities will also change

TestPlan should:
* convert the TestSet to a set of functionality to execute

TestRunner should:
* have the option to run synchronously and asynchronously (serial / parallel)
* pass results to the output writer

**Comment by jamesrichford** [comment]
Don't forget to refactor out TestFixture.getTests as this functionality (filtering focused tests should be handled by the creation of a test plan)

**Comment by Jameskmonger** [comment]
Resolved by #130
